extensions/scarb-doc/src/lib.rs

```diff
@@ -48,16 +73,22 @@ pub fn generate_packages_information(
             },
         });
     }
-    packages_information
+    Ok(packages_information)
 }
 
 fn generate_language_elements_tree_for_package(
     package_name: String,
     project_config: ProjectConfig,
+    features_cfg: CfgSet,
 ) -> Crate {
-    let db = ScarbDocDatabase::new(Some(project_config));
+    let db: ScarbDocDatabase =
+        ScarbDocDatabase::new(&package_name, Some(project_config), features_cfg);
 
     let main_crate_id = db.intern_crate(CrateLongId::Real(package_name.into()));
 
     Crate::new(&db, main_crate_id)
 }
+
+fn initial_cfg_set() -> CfgSet {
+    CfgSet::from_iter([Cfg::name("doc")])
+}
```

Comment:
Should be in `db.rs` imo and hidden in db constructor, since we always want to extend the set with `doc`